BraveWallet/Crypto/Transactions/TransactionDetailsView.swift

```diff
@@ -0,0 +1,264 @@
+      return (value, {
+        guard let doubleValue = Double(value), let assetRatio = assetRatios[networkStore.selectedChain.symbol.lowercased()] else {
+          return "$0.00"
+        }
+        return numberFormatter.string(from: NSNumber(value: doubleValue * assetRatio)) ?? "$0.00"
+      }())
+    }
+    return nil
+  }
+  
+  /// The formatted transaction fee for the transaction, ex. `"0.003401\n$15.57"`
+  private var transactionFee: String? {
+    guard let (fee, fiat) = gasFee else {
+      return nil
+    }
+    let symbol = networkStore.selectedChain.symbol
+    return String(format: "%@ %@\n%@", fee, symbol, fiat)
+  }
+  
+  /// The market price for the asset
+  private var marketPrice: String {
+    let symbol = networkStore.selectedChain.symbol
+    let marketPrice = numberFormatter.string(from: NSNumber(value: assetRatios[symbol.lowercased(), default: 0])) ?? "$0.00"
+    return marketPrice
+  }
+
+  private func namedAddress(for address: String) -> String {
+    NamedAddresses.name(for: address, accounts: keyringStore.keyring.accountInfos)
+  }
+
+  private var title: String {
+    switch info.txType {
+    case .erc20Approve:
+      return Strings.Wallet.transactionUnknownApprovalTitle
+    default:
+      if info.isSwap {
+        return Strings.Wallet.swap
+      } else {
+        return Strings.Wallet.sent
+      }
+    }
+  }
+
+  private var header: some View {
+    TransactionHeader(
+      fromAccountAddress: info.fromAddress,
+      fromAccountName: namedAddress(for: info.fromAddress),
+      toAccountAddress: info.ethTxToAddress,
+      toAccountName: namedAddress(for: info.ethTxToAddress),
+      transactionType: title,
+      value: value,
+      fiat: fiat
+    )
+      .frame(maxWidth: .infinity)
+      .padding(.vertical, 30)
+  }
+  
+  var body: some View {
+    NavigationView {
+      List {
+        Section(
+          header: header
+            .resetListHeaderStyle()
+            .osAvailabilityModifiers { content in
+              if #available(iOS 15.0, *) {
+                content // Padding already applied
+              } else {
+                content.padding(.top)
+              }
+            }
+        ) {
+          if let transactionFee = transactionFee {
+            detailRow(title: Strings.Wallet.transactionDetailsTxFeeTitle, value: transactionFee)
+          }
+          detailRow(title: Strings.Wallet.transactionDetailsMarketPriceTitle, value: marketPrice)
+          detailRow(title: Strings.Wallet.transactionDetailsDateTitle, value: dateFormatter.string(from: info.createdTime))
+          Button(action: {
+            if let baseURL = self.networkStore.selectedChain.blockExplorerUrls.first.map(URL.init(string:)),
+               let url = baseURL?.appendingPathComponent("tx/\(info.txHash)") {
+              openWalletURL?(url)
+            }
+          }) {
+            detailRow(title: Strings.Wallet.transactionDetailsTxHashTitle) {
```

Comment:
Wrap this in a check so that you only show the button if the tx hash is non-empty (that way you can remove the *** below)

Reply:
This was done to match desktop behaviour
<img width="312" alt="Screen Shot 2022-03-17 at 5 46 14 PM" src="https://user-images.githubusercontent.com/5314553/158900120-32449735-3ffe-4330-b9b3-bce9e431ae40.png">